### ISSUE TITLE
On-screen keyboard controls

### ISSUE THREAD

**Issue by Chikorita-Lover** [ScratchAddons/ScratchAddons#5148]
### Why this would be helpful/solve a problem

Sometimes, I want to play Scratch games that use key controls on a mobile platform.

### What this addon does

This addon adds a button (like the gamepad addon) to configure on-screen controls. While Scratch Addons can't be used on mobile, it can be used for touch-screen  computers, or used on TurboWarp on mobile devices.

**Comment by BroJac5246** [comment]
Good idea! We could have a keyboard button in, say, the bottom left, that would bring up a virtual one. I also think it would be cool if you could have different arrangements or mappings, so that you could have, say, only the WASD and arrow keys showing.

**Comment by Chikorita-Lover** [comment]
Yeah! What I was thinking was being able to add buttons and joysticks that correlate to different keys on the keyboard. The joysticks would work just like the gamepad addon, and you could position the buttons/joysticks as wanted.

**Comment by BroJac5246** [comment]
> Yeah! What I was thinking was being able to add buttons and joysticks that correlate to different keys on the keyboard. The joysticks would work just like the gamepad addon, and you could position the buttons/joysticks as wanted.

That could be a really interesting way to control a project, even if you don't need touchscreen controls. It would almost be like adding buttons to the project if we implement it right.

**Comment by Secret-chest** [comment]
This would be PERFECT for portrait mode.

**Comment by rgantzosonscratch** [comment]
This doesn’t sound very necessary. Most computers already have this built in as an accessibility feature- especially touchscreen computers.

**Comment by Chikorita-Lover** [comment]
It would be very useful for mobile players using TurboWarp, however.

**Comment by BroJac5246** [comment]
> This doesn’t sound very necessary. Most computers already have this built in as an accessibility feature- especially touchscreen computers.

Not all devices would let you do that. Say you're on a Chromebook stuck in tablet mode. You need a Scratch project to detect if a key is pressed. Unless you attach an external keyboard or go to some complicated lengths, you can't.

Also, even if your device _does_ let you do that through accessibility, do you actually want to go to the pains of dealing with it?

**Comment by WorldLanguages** [comment]
It would be complex to support the keyboard for every language, but Scratch doesn't support any of those characters anyway right?

**Comment by mxmou** [comment]
> It would be complex to support the keyboard for every language, but Scratch doesn't support any of those characters anyway right?

I think it's possible to detect them using the join block workaround.

**Comment by Secret-chest** [comment]
> It would be complex to support the keyboard for every language,

Do we need that? The obvious use for this would be to control games.

**Comment by WorldLanguages** [comment]
Related to #7012
